include/ccl_background.h

```diff
@@ -243,6 +261,17 @@ double ccl_scale_factor_of_chi(ccl_cosmology * cosmo, double chi, int * status);
  */
 void ccl_scale_factor_of_chis(ccl_cosmology * cosmo, int nchi, double chi[], double output[], int* status);
 
+/**
+ * Physical density (rho) as a function of scale factor.  Critical density is defined as rho_critical = 3 H^2(a)/ (8 pi G). Density of a given species is then rho_x = Omega_x(a) rho_critical(a). For example, rho_matter(a) = Omega_m  a^{-3} / (H^2/H0^2)  3H^2 / (8 pi G) =  Omega_m  a^{-3}  3H0^2 / (8 pi G) =  Omega_m a^{-3}  rho_critical_present. Units of M_sun/(Mpc)^3.
+ * @param cosmo Cosmological parameters
+ * @param a scale factor, normalized to 1 for today
+ * @param label species type. Available: 'critical'(0,1), 'matter'(2,3), 'dark_energy'(4,5), 'radiation'(6,7), 'curvature'(8,9), 'massless neutrinos'(10,11), 'massive neutrinos'(12,13). Even label numbers refer to the physical densities. Odd label numbers refer to 'comoving' densities, which multiply the physical densities by a^3.
+ * @param status Status flag. 0 if there are no errors, nonzero otherwise. 
+ * For specific cases see documentation for ccl_error.
+ * @return rho_x, physical density at scale factor a. 
+ */
+double ccl_rho_x(ccl_cosmology * cosmo, double a, ccl_rho_x_label label, int* status);
```

Comment:
here you could use ccl_omega_x_label instead of ccl_rho_x_label. The former does not have "comoving" options, but you could just add one more parameter "is_comoving" that can be 0 or 1 depending on whether the user wants the comoving density. I'm pretty sure this would actually simplify the code a lot too (by a factor of 2 in number of lines ;-))